src/Interpreters/InterpreterInsertQuery.cpp

```diff
@@ -56,7 +56,7 @@ namespace ErrorCodes
 
 InterpreterInsertQuery::InterpreterInsertQuery(
     const ASTPtr & query_ptr_, const Context & context_, bool allow_materialized_, bool no_squash_, bool no_destination_)
-    : query_ptr(query_ptr_)
+    : query_ptr(query_ptr_->clone())
```

Comment:
>weired test failure.

```
Code: 81, e.displayText() = DB::Exception: Database name is empty (version 20.13.1.5584)
```

The problem is that InterpreterInsertQuery modifies the query, it resolves the storage id (table, database), see https://github.com/ClickHouse/ClickHouse/blob/95ac01ab829df234cc59e91b94ac9e2ef097e3e1/src/Interpreters/InterpreterInsertQuery.cpp#L78

W/o clone it modified the passed query, after this change it modified copied version and so `InputStreamFromASTInsertQuery` receives empty database.

Also I noticed that by some reason `DatabaseCatalog::getTableImpl` does not respect `current_database` (from `Context`), maybe @tavplubix can answer this.

```patch
diff --git a/src/Interpreters/DatabaseCatalog.cpp b/src/Interpreters/DatabaseCatalog.cpp
index e9caaab4ef..a263846b14 100644
--- a/src/Interpreters/DatabaseCatalog.cpp
+++ b/src/Interpreters/DatabaseCatalog.cpp
@@ -250,12 +250,16 @@ DatabaseAndTable DatabaseCatalog::getTableImpl(
 
     DatabasePtr database;
     {
+        std::string database_name = context.getCurrentDatabase();
+        if (!table_id.database_name.empty())
+            database_name = table_id.getDatabaseName();
+
         std::lock_guard lock{databases_mutex};
-        auto it = databases.find(table_id.getDatabaseName());
+        auto it = databases.find(database_name);
         if (databases.end() == it)
         {
             if (exception)
-                exception->emplace(ErrorCodes::UNKNOWN_DATABASE, "Database {} doesn't exist", backQuoteIfNeed(table_id.getDatabaseName()));
+                exception->emplace(ErrorCodes::UNKNOWN_DATABASE, "Database {} doesn't exist", backQuoteIfNeed(database_name));
             return {};
         }
         database = it->second;
```

Although `InterpreterInsertQuery::getTable` do more then just this, so I guess it should be moved into the upper scope (or copied into the `InputStreamFromASTInsertQuery`)